### PR TITLE
SP-674 remove unnessecary backend calls from healthcheck 

### PR DESCRIPTION
**Pull request by tomsmyers**
this PR also includes:
* a one-liner to fix hanging test workers caused by delayed `nock` server responses.
* refactoring of `healthCheckFactory` to make adding and removing services more robust in the future.

### CODE REVIEW

File: common/config.js [hunk 1]
```diff
@@ -19,7 +19,6 @@ function get(name, fallback, options = {}) {
 }
 
 module.exports = {
-  sessionSecret: get('SESSION_SECRET', 'app-insecure-default-session', { requireInProduction: true }),
   apis: {
     oauth2: {
```
Comment (jonhindle) [inline common/config.js:23]: Could I get you to rip out oauth2 and offenderAssessment whilst you are there, please? 😬

File: common/services/healthcheck.js [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
   return () =>
     check()
       .then(result => ({ name, status: 'ok', message: result }))
       .catch(err => ({ name, status: 'ERROR', message: err }))
 }
 
-module.exports = (auth, sentencePlanning, elite2) => {
-  const checks = [service('auth', auth), service('sentencePlanning', sentencePlanning), service('elite2', elite2)]
+module.exports = (...services) => {
+  const checks = services.map(s => service(s.name, s.config))
```
Comment (jonhindle) [inline common/services/healthcheck.js:12]: de-struct in the arguments:
const checks = services.map(({name, config}) => service(name, config))

Comment (jonhindle) [reply common/services/healthcheck.js:12]: I much prefer the services object though 👍

File: helm_deploy/values-preprod.yaml [hunk 3]
```diff
@@ -24,17 +24,14 @@ ingress:
   host: preprod.sentence-planning.service.justice.gov.uk
   path: /
   cert_secret: sentence-planning-cert
-  
+
 env:
-  NOMIS_AUTH_URL: https://gateway.preprod.nomis-api.service.hmpps.dsd.io/auth
-  NOMIS_OAUTH_PUBLIC_KEY: LS0tLS1CRUdJTiBQVUJMSUMgS0VZLS0tLS0KTUlJQklqQU5CZ2txaGtpRzl3MEJBUUVGQUFPQ0FROEFNSUlCQ2dLQ0FRRUFqVGR1dEtLZVBSQmVBY2tnUzZwLwpsVitkNVlxYjNxMTh6NlV3aDFNTll4cDNIcVlwWVYrOTJ5RnhlMlE1UjdLVmpJeUt5MmZHQWpzV3krZVNOMjNUCkZURVVUeXdHdnc0SkJrRGRoaUUvKy80SjR2TWFHMzBYN29GdE9JUWIwaENFQVlWM3o4emc1WVd2OGZlRkVnWGcKSndKNUJzaStTQ29qM0hkOURFeTVkZlpNOWMvcGg5c3lhaktTVmJBbytEK1dqeFBYdThSd1FKT0FTSDQ1RXByVAozSTMvTXZZcmNWb3ZDeFFWclFjcGMzYmNFNGx6WHRZUlcvcElqR1FaQSsvVmIzV2V2bkdRYmJ3UzR6enl5M2s4CkRCZ0pzQVJlVDI1eHp0ckptRGlZZXBHYkVKVHV0alRqR2NVOG1qUjRWZ3gwdHdqMk9DMzBES3dXbGlrMzhXTWwKTlFJREFRQUIKLS0tLS1FTkQgUFVCTElDIEtFWS0tLS0t
-  ELITE2_API_URL: https://gateway.preprod.nomis-api.service.hmpps.dsd.io/elite2api
   OASYSAPI_ENDPOINT_URL: https://offender-preprod.aks-live-1.studio-hosting.service.hmpps.dsd.io
```
Comment (jonhindle) [inline helm_deploy/values-preprod.yaml:29]: Could you rip this out too whilst you're here please 😬

File: helm_deploy/values-prod.yaml [hunk 4]
```diff
@@ -26,15 +26,12 @@ ingress:
   cert_secret: sentence-planning-cert
 
 env:
-  NOMIS_AUTH_URL: https://gateway.prod.nomis-api.service.hmpps.dsd.io/auth
-  NOMIS_OAUTH_PUBLIC_KEY: LS0tLS1CRUdJTiBQVUJMSUMgS0VZLS0tLS0KTUlJQklqQU5CZ2txaGtpRzl3MEJBUUVGQUFPQ0FROEFNSUlCQ2dLQ0FRRUFwSkdtYWp6SW9POVdSY2owM2VTbQpFeWJTYnNJaEdkNkRWVDRxUWFSRTZwN3RIN1ppSklvRWtzczRkOFphMGhuYWJwa0JqWTYwS3NnMW5teXpoZVlSCithWXVtc0gxbm1JcmJ4ZCtqV043T3A2V1pteWRBU3JMek1XWVdBYlNHOHBXMXkxY1RwdkRaMTlBSGJTYTRRL2UKSnRlcDFscCsyMnVHMW5kTVRwclRkNjRLTkcyZnRwcmV3aFExeE5xdmtUWjZoQnVRRVJKMzFhUXBkRThNQTFxSwpDWnI3VDJxTk1BTWNkTlpia0FqRjVLOXE0alZ0QkY3WXFPeXJ0T1JmTldqYUJoWHA5TlN0YWJCTXBURTdQNm8vClhJaU9rcmltNjlzaG1OdDhVRlU3WUlJN0ZvejgzM29HZnk2dmtDRjRxcGpIZ0RNc21STjc1UXRrRmRtaU8xaDgKcHdJREFRQUIKLS0tLS1FTkQgUFVCTElDIEtFWS0tLS0tCg==
-  ELITE2_API_URL: https://gateway.prod.nomis-api.service.hmpps.dsd.io/elite2api
   OASYSAPI_ENDPOINT_URL: https://dev.devtest.assessment-api.hmpps.dsd.io
```
Comment (jonhindle) [inline helm_deploy/values-prod.yaml:29]: Could you rip this out too whilst you're here please 😬

File: common/config.js (hunk 1 above)
Comment (tomsmyers) [reply common/config.js]: done

File: common/services/healthcheck.js (hunk 2 above)
Comment (tomsmyers) [reply common/services/healthcheck.js]: done

File: helm_deploy/values-prod.yaml (hunk 4 above)
Comment (tomsmyers) [reply helm_deploy/values-prod.yaml]: done

**Pull request merged**